<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html><head><title>LDIF - Benchmark Results</title>



    
    <style>
        body { background: white; color: black; font-family: sans-serif; line-height: 1.4em; padding: 3em 12em 3em 12em; margin: 0; }
        :link { color: #00c; }
        :visited { color: #609; }
        a:link img { border: none; }
        a:visited img { border: none; }
        img {
            margin: 2em 0 2em 4em;
        }
        h1, h2, h3 { background: white; color: #800; }
        h1 { font: 170% sans-serif; margin: 0; }
        h2 { clear: both; font: 140% sans-serif; margin: 2.5em 0 -0.5em 0; padding-top: 1.5em; border-top: 1px solid #ccc;}
        h3 { font: 115% sans-serif; margin: 2em 0 -0.5em 0; }
        h4 { font: bold 100% sans-serif; }
        h5 { font: italic 100% sans-serif; }
        h6 { font: small-caps 100% sans-serif; }
        .hide { display: none; }
        pre { background: #fff6bb; font-family: monospace; line-height: 1.2em; padding: 1em 2em; }
        dt { font-weight: bold; margin-top: 0; margin-bottom: 0; }
        dd { margin-top: 0; margin-bottom: 0; }
        code, tt { font-family: monospace; }
        ol.toclist li a { text-decoration: none; }
        .note { color: red; }
        #header { border-bottom: 1px solid #ccc; }
        #logo { float: right; }
        #authors { clear: right; float: right; font-size: 80%; text-align: right;}
        #toc {  float: left; }
        #content { clear: both; margin: 2em auto 0 0; text-align: justify }
        #download h2, #demo h2 { font-size: 125%; margin: 1.5em 0 -0.2em 0; }
        #download small, #demo small { color: #888; font-size: 80%; }
        #footer { border-top: 1px solid #ccc; color: #aaa; margin: 2em 0 0; }
        @media Print {
            * { font-size: 92%; }
            body { padding: 0; line-height: 1.2em; }
            #content { margin: 0; width: 100%; }
        }
        @media Aural {
            h1 { stress: 20; richness: 90; }
            h2 { stress: 20; richness: 90; }
            h3 { stress: 20; richness: 90; }
            .hide { speak: none; }
            dt { pause-before: 20%; }
            pre { speak-punctuation: code; }
        }
        .Stil1 {color: #FF0000}
        table {border-collapse: collapse; margin-bottom: .75em; border-top: 1px solid #D4DCE8; border-bottom: 1px solid #D4DCE8; margin-left: 10%;}
        table th {text-align: right; background: #EAF3FA; padding: .5em .9em; height: 21px; font-weight: bold; font-size: 1em; color: #333}
        table.results th {text-align: left; background: #EAF3FA; padding: .5em .9em; height: 21px; font-weight: bold; font-size: 1em; color: #333}
        table th a {text-decoration: none !important}
        table td {padding: .5em 1em; border-bottom: 1px solid #eee}
        table.num td {vertical-align: top; text-align: right;}
        table tr.even {background-color: #fff}
        table tr.odd {background-color: #f6f6f6}
    </style>

    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"></head>

<body>

<div id="logo" align="right"><a href="http://wbsg.de">
    <img src="images/fu-logo.gif" alt="Freie Universität Berlin Logo" border="0"></a><br/>
    <a href="http://mes-semantics.com">
        <img src="images/mes-semantics.png" alt="mes|semantics"></a>
</div>

<div id="header">
    <h1 style="font-size: 250%;">LDIF - Benchmark Results<br></h1>
</div>

<div id="authors">
    <a href="http://www.wiwiss.fu-berlin.de/en/institute/pwo/bizer/team/SchultzAndreas.html">Andreas Schultz</a><br>
    <a href="mailto:a.matteini@mes-semantics.com">Andrea Matteini</a><br>
    <a href="http://www.wiwiss.fu-berlin.de/en/institute/pwo/bizer/team/IseleRobert.html">Robert Isele</a><br>
    <a href="http://www.wiwiss.fu-berlin.de/en/institute/pwo/bizer/team/BizerChristian.html">Chris Bizer</a><br>
    <a href="mailto:c.becker@mes-semantics.com">Christian Becker</a><br>
</div>

<div id="toc">
<h2 id="contents">Contents</h2>
<ul class="toclist">


    <li><a href="#intro">Introduction</a></li>
    <li><a href="#useCase">Use Case</a></li>
    <ul>
        <li><a href="#datasets">Datasets</a></li>
        <li><a href="#mappings">Mappings</a></li>
        <li><a href="#linkspecs">Link Specififcation</a></li>
    </ul>
    <li><a href="#machine">Benchmark Machines</a></li>
    <li><a href="#procedure">Test Procedure</a></li>

    <li><a href="#results">Results</a></li>
</ul>
</div>

<h2 id="intro">Introduction</h2>
<p> This page presents a performance evaluation for <a href="http://ldif.wbsg.de">LDIF</a> using life science use cases.</p>

<p>The following data sources have been used to evaluate the performance of LDIF:</p>
<ul>
  <li>
      <a href="http://mouse.brain-map.org/">Allen Mouse Brain Atlas</a> - a growing collection of online public resources integrating extensive gene expression and neuroanatomical data;
  </li>
  <li>
      <a href="http://www.genome.jp/kegg/genes.html">KEGG GENES</a> - a collection of gene catalogs for all complete genomes generated from publicly available resources;
  </li>
  <li>
      <a href="http://www.genome.jp/kegg/pathway.html">KEGG Pathway</a> - a collection of pathway maps representing knowledge on the molecular interaction and reaction networks;
  </li>
  <li>
      <a href="http://pharmgkb.org/">PharmGKB</a> - which provides data on gene information, disease and drug pathways, and SNP variants;
  </li>
  <li>
      <a href="http://www.uniprot.org/">Uniprot</a> - a dataset containing protein sequence, genes and function.
  </li>
</ul>





<h2 id="useCase"> Use case</h2>
<p>
    In this use case, we evaluated the performance of LDIF integrating two larger life science datasets and translate those into a common <a href="resources/Wiki.owl">target vocabulary</a>.
</p>


<h3 id="datasets">Datasets</h3>

<p>
    For this use case we used only KEGG GENES and UniProt datasets.
    There is a huge difference in dataset size between the two datasets.
    Converted to N-Triples the complete KEGG GENES dump is about 28GB in size whereas the UniProt dataset contains over 400GB worth of data.
</p>

<p>
    For the test, we generated subsets of both data sources amounting together to 25, 100, 150 and 300 million RDF triples.
    The 3690 M dataset include the complete UniProt and complete KEGG GENES datasets.
</p>

<p>
    Details about the benchmark datasets are summarized in the following table.
    It provides statistics about the data integration process for each dataset.
    The original number of input triples decreases in the process as LDIF discards input triples which are irrelevant for the defined mappings, and therefore can not be translated into the target vocabulary.
    The number decreases again after the actual translation, as the input data uses more verbose vocabularies and as multiple triples from the input data are thus combined into single triples in the target vocabulary.
    The size of the final dataset is the number of quads after the mapping phase plus any provided provenance data.
</p>

<table class="num">
    <tbody>
        <tr>
            <td>   </td>
            <th> 25M</th>
            <th> 100M</th>
            <th> 150M</th>
            <th> 300M</th>
            <th> 3690M</th>
        </tr>
        <tr>
            <th>Number of input quads</th>
            <td> 25,000,000 </td>
            <td> 100,000,000 </td>
            <td> 150,000,000 </td>
            <td> 300,000,000 </td>
            <td> 3,687,918,681 </td>
        </tr>
        <tr>
            <th>Number of quads after irrelevance filter</th>
            <td> 13,576,394</td>
            <td> 44,249,757</td>
            <td> - </td>
            <td> - </td>
            <td> 1,164,250,713 </td>
        </tr>
        <tr>
            <th>Number of quads after mapping</th>
            <td> 4,419,410</td>
            <td> 24,972,112</td>
            <td> - </td>
            <td> - </td>
            <td> 98,380,001 </td>
        </tr>
        <tr>
            <th>Number of pairs of equivalent entities resolved</th>
            <td> 24,782</td>
            <td> 213,062</td>
            <td> - </td>
            <td> - </td>
            <td> 6,321,750 </td>
        </tr>
        <tr>
            <th>Overall file size</th>
            <td> 5.6 GB</td>
            <td> 23 GB</td>
            <td> - </td>
            <td> - </td>
            <td> - </td>
        </tr>
         <tr>
            <th>Download link</th>
            <td> <a href="http://www5.wiwiss.fu-berlin.de/ldif/resources/usecases/25m.zip">25M.zip</a> </td>
            <td> <a href="http://www5.wiwiss.fu-berlin.de/ldif/resources/usecases/100m.zip">100M.zip</a> </td>
            <td> - </td>
            <td> - </td>
            <td> - </td>
        </tr>
    </tbody>
</table>


<h3 id="mappings">Mappings</h3>

<p>
    We defined <a href="resources/mappings.ttl"> R2R mappings</a> for translating genes, diseases and pathways from KEGG GENES and genes from UniProt into a proprietary target vocabulary.
    Some more sophisticated mappings from the use case translate complex structural patterns and perform value transformations (e.g. extracting an integer value from a URI).
    The prevalent value transformations are extracting strings with a regular expression and modifying the target data types.
</p>

Here are some examples of the mappings we used for:
<ul>
    <li> mapping a KEGG GENE <tt>gene</tt> into a target vocabulary <tt>Gene</tt>
        <pre><code> mp:Gene<br>   a r2r:ClassMapping;<br>   r2r:prefixDefinitions<br>        "category: &lt;http://mywiki/resource/category/&gt; .<br>         property: &lt;http://mywiki/resource/property/&gt; .<br>         pathway:  &lt;http://wiking.vulcan.com/neurobase/kegg_pathway/resource/vocab/&gt; .<br>         genes:    &lt;http://wiking.vulcan.com/neurobase/kegg_genes/resource/vocab/&gt; .<br>         xsd:      &lt;http://www.w3.org/2001/XMLSchema#&gt; .";<br>   r2r:sourcePattern "?SUBJ a genes:gene";<br>   r2r:targetPattern "?SUBJ a category:Gene";<br>   .</code></pre>
    </li>
    <li> mapping a KEGG GENE <tt> externalLink</tt>  property into a target vocabulary <tt> UniprotId</tt>  property
        <pre><code> mp:GeneLinkUniProt<br>   a r2r:PropertyMapping;<br>   r2r:mappingRef mp:Gene;<br>   r2r:sourcePattern  "?SUBJ genes:externalLink ?x";<br>   r2r:transformation "?id = regexToList('UniProt:(.+)', ?x)";<br>   r2r:targetPattern  "?SUBJ property:UniprotId ?'id'^^xsd:string";<br>   . </code></pre>
    </li>
</ul>




<h3 id="linkspecs">Link Specifications</h3>
<p>
    We defined the following <a href="resources/linkSpecs.xml">Link Specification</a> for the identity resolution phase:
</p>
<pre><code>&lt;Silk&gt;
	&lt;Prefixes&gt;
		&lt;Prefix id="rdf" namespace="http://www.w3.org/1999/02/22-rdf-syntax-ns#" /&gt;
		&lt;Prefix id="rdfs" namespace="http://www.w3.org/2000/01/rdf-schema#" /&gt;
		&lt;Prefix id="owl" namespace="http://www.w3.org/2002/07/owl#" /&gt;
		&lt;Prefix id="property" namespace="http://mywiki/resource/property/" /&gt;
		&lt;Prefix id="category" namespace="http://mywiki/resource/category/" /&gt;
	&lt;/Prefixes&gt;

	&lt;Interlinks&gt;
		&lt;Interlink id="link"&gt;<br>			&lt;LinkType&gt;owl:sameAs&lt;/LinkType&gt;<br><br>			&lt;SourceDataset dataSource="Source" var="a"&gt;<br>				&lt;RestrictTo&gt;<br>					{ ?a rdf:type category:Gene }<br>					UNION { ?a rdf:type category:Disease }<br>					UNION { ?a rdf:type category:Pathway }<br>				&lt;/RestrictTo&gt;<br>			&lt;/SourceDataset&gt;<br><br>			&lt;TargetDataset dataSource="Target" var="b"&gt;<br>				&lt;RestrictTo&gt;<br>					{ ?b rdf:type category:Gene }<br>					UNION { ?b rdf:type category:Disease }<br>					UNION { ?b rdf:type category:Pathway }<br>				&lt;/RestrictTo&gt;<br>			&lt;/TargetDataset&gt;<br><br>			&lt;LinkageRule&gt;<br>			&lt;Aggregate type="max"&gt;<br>					&lt;Compare metric="equality"&gt;<br>						&lt;Input path="?a/property:UniprotId" /&gt;<br>						&lt;Input path="?b/property:UniprotId" /&gt;<br>					&lt;/Compare&gt;<br>					&lt;Compare metric="equality"&gt;<br>						&lt;Input path="?a/property:EntrezGeneId" /&gt;<br>						&lt;Input path="?b/property:EntrezGeneId" /&gt;<br>					&lt;/Compare&gt;<br>					&lt;Compare metric="equality"&gt;<br>						&lt;Input path="?a/property:MgiMarkerAccessionId" /&gt;<br>						&lt;Input path="?b/property:MgiMarkerAccessionId" /&gt;<br>					&lt;/Compare&gt;<br>					&lt;Compare metric="equality"&gt;<br>						&lt;Input path="?a/property:KeggDiseaseId" /&gt;<br>						&lt;Input path="?b/property:KeggDiseaseId" /&gt;<br>					&lt;/Compare&gt;<br>					&lt;Compare metric="equality"&gt;<br>						&lt;Input path="?a/property:KeggPathwayId" /&gt;<br>						&lt;Input path="?b/property:KeggPathwayId" /&gt;<br>					&lt;/Compare&gt;<br>			&lt;/Aggregate&gt;<br>			&lt;/LinkageRule&gt;<br><br>			&lt;Filter threshold="1.0" /&gt;<br><br>		&lt;/Interlink&gt;<br><br>	&lt;/Interlinks&gt;<br>&lt;/Silk&gt;</code>
</pre>




<h2 id="machine">Benchmark Machines</h2>
<p>We used a machine with the following specification for the benchmark experiments:</p>

<h3>FUB machines</h3>
<p>
    These machines have been used for In-Memory, TDB and FUB Hadoop clusters.
</p>
<table>
    <tr>
        <td>Hardware</td>
        <td>
            Processors: Intel i7 950, 3.07GHz (quadcore)
            <br/>
            Memory: 24GB
            <br/>
            Hard Disks: 2 × 1.8TB (7,200 rpm) SATA2
        </td>
    </tr>
    <tr>
        <td>Software</td>
        <td>
            Operating System: Ubuntu 11.04 64-bit, Kernel: 2.6.38-10
            <br/>
            Java version: 1.6.0_22
        </td>
    </tr>
</table>

<h3>EC2 c1.medium instances</h3>

<p>
    There machines have been used for EC2 Hadoop clusters.
</p>
<table>
    <tr>
        <td>Hardware</td>
        <td>Processor: 5 EC2 Compute Units<sup><a href="#fn1">1</a></sup>
            <br/>
            Memory: 1.7 GB
            <br/>
            Hard Disks: 350 GB
            <br/>
            I/O Performance: Moderate
        </td>
    </tr>
    <tr>
        <td>Software</td>
        <td>
            Operating System: Ubuntu 11.04 32-bit
        </td>
    </tr>
</table>


<h3>Hadoop clusters</h3>

<p>In all cluster configurations, the master works as job tracker and name node, while the slaves work as data nodes and task trackers.</p>

<p>FUB Hadoop 2-slaves cluster
<ul>
    <li> 1 master, 2 slaves (FUB machines) </li>
    <li> Network: Gigabit Ethernet </li>
</ul>
</p>

<p>EC2 Hadoop X-slaves cluster
<ul>
    <li> 1 master,  X slaves (EC2 c1.medium instances)  </li>
</ul>
</p>


<h2 id="procedure">Test Procedure</h2>
<p>
    We used LDIF version 0.5 both for the in-memory, TDB and Hadoop tests.
</p>
For each single machine test, we applied the following procedure:
</p>
<ol>
    <li>Clear Operating System caches
        <pre> echo 2 &gt; /proc/sys/vm/drop_caches </pre>
    </li>
    <li>Run test
        <pre> java -server -Xmx20G -jar ldif-single-machine.jar </pre>
    </li>
</ol>

<p>
    Details about the Hadoop cluster configurations are available at <a href="https://github.com/wbsg/ldif/wiki/Hadoop-configuration">this page</a>.
</p>





<h2 id="results">Results</h2>

<p>The following table summarizes the LDIF run times for the different dataset sizes.
    The overall run time is split according to the different processing steps of the integration process. </p>

<h3>25M run times</h3>

<img src="resources/charts/25M.png" />

<table class="results">
    <tr>
        <th>Phase </th>
        <th>In-memory  </th>
        <th>TDB  </th>
        <th>Hadoop FUB 2-slaves </th>
    </tr>
    <tr>
        <td> Load and build entities for R2R </td>
        <td>186.3 s</td>
        <td> 1334 s</td>
        <td> 688 s</td>
    </tr>
    <tr>
        <td> R2R data translation </td>
        <td> 116.4 s </td>
        <td> 85 s </td>
        <td> 87 s</td>
    </tr>
    <tr>
        <td> Build entities for Silk </td>
        <td>12.9  s </td>
        <td> 139 s </td>
        <td> 225 s</td>
    </tr>
    <tr>
        <td> Silk identity resolution </td>
        <td>72.6 s</td>
        <td> 213 s</td>
        <td> 216 s</td>
    </tr>
    <tr>
        <td> URIs rewriting </td>
        <td>16.2 s</td>
        <td>15 s</td>
        <td> 427 s</td>
    </tr>
    <tr>
        <td> Overall execution  </td>
        <td> 6.7 min</td>
        <td> 29.7 min </td>
        <td> 27.4 min</td>
    </tr>
</table>





<h3>100M run times</h3>

<img src="resources/charts/100M.png" />

<table class="results">
    <tr>
        <th>Phase </th>
        <th>In-memory </th>
        <th>TDB  </th>
        <th>Hadoop FUB 2-slaves </th>
    </tr>
    <tr>
        <td> Load and build entities for R2R </td>
        <td>822 s</td>
        <td> 7014 s</td>
        <td>3112 s</td>
    </tr>
    <tr>
        <td> R2R data translation </td>
        <td>744 s</td>
        <td> 607 s</td>
        <td>165 s</td>
    </tr>
    <tr>
        <td> Build entities for Silk </td>
        <td>81 s</td>
        <td> 806 s</td>
        <td>405 s</td>
    </tr>
    <tr>
        <td> Silk identity resolution </td>
        <td>772 s</td>
        <td>4656 s</td>
        <td>544 s</td>
    </tr>
    <tr>
        <td> URIs rewriting </td>
        <td>113  s</td>
        <td> 118 s</td>
        <td> 1056 s</td>
    </tr>
    <tr>
        <td> Overall execution  </td>
        <td>42.2 min </td>
        <td> 220 min</td>
        <td> 88 min</td>
    </tr>
</table>



<h3>150M run times</h3>


<img src="resources/charts/150M.png" />

<table class="results">
    <tr>
        <th>Phase </th>
        <th>In-memory  </th>
        <th>TDB  </th>
        <th> Hadoop FUB 2-slaves </th>
    </tr>
    <tr>
        <td> Load and build entities for R2R </td>
        <td>Out of Memory</td>
        <td> 13776 s </td>
        <td> 3830 s</td>
    </tr>
    <tr>
        <td> R2R data translation </td>
        <td>- </td>
        <td>1206 s </td>
        <td>170 s</td>
    </tr>
    <tr>
        <td> Build entities for Silk </td>
        <td>- </td>
        <td>847 s </td>
        <td>380 s</td>
    </tr>
    <tr>
        <td> Silk identity resolution </td>
        <td>- </td>
        <td>5328 s </td>
        <td> 688 s </td>
    </tr>
    <tr>
        <td> URIs rewriting </td>
        <td>- </td>
        <td> 173 s</td>
        <td> 1235 s</td>
    </tr>
    <tr>
        <td> Overall execution  </td>
        <td>- </td>
        <td>355 min </td>
        <td>105 min </td>
    </tr>
</table>




<h3>300M run times</h3>


<img src="resources/charts/300M.png" />

<table class="results">
    <tr>
        <th>Phase </th>
        <th>In-memory  </th>
        <th>TDB  </th>
        <th>Hadoop FUB 2-slaves </th>
    </tr>
    <tr>
        <td> Load and build entities for R2R </td>
        <td> Out of Memory </td>
        <td> 22870 s</td>
        <td> 6070 s</td>
    </tr>
    <tr>
        <td> R2R data translation </td>
        <td>- </td>
        <td>1203 s </td>
        <td> 179 s </td>
    </tr>
    <tr>
        <td> Build entities for Silk </td>
        <td>- </td>
        <td>1006 s </td>
        <td> 436 s </td>
    </tr>
    <tr>
        <td> Silk identity resolution </td>
        <td>- </td>
        <td>8392 s </td>
        <td>1022 s </td>
    </tr>
    <tr>
        <td> URIs rewriting </td>
        <td>- </td>
        <td> 176 s</td>
        <td> 1232 s</td>
    </tr>
    <tr>
        <td> Overall execution  </td>
        <td>- </td>
        <td>560 min </td>
        <td>148 min </td>
    </tr>
</table>




<h3>3690M run times</h3>

<img src="resources/charts/3690M.png" />

<table class="results">
    <tr>
        <th>Phase </th>
        <th>Hadoop FUB 2-slaves <br/>(format: hh:mm:ss)</th>
    </tr>
    <tr>
        <td> Load and build entities for R2R </td>
        <td style="text-align:right;">9:34:44 </td>
    </tr>
    <tr>
        <td> R2R data translation </td>
        <td style="text-align:right;">15:22 </td>
    </tr>
    <tr>
        <td> Build entities for Silk </td>
        <td style="text-align:right;">51:12 </td>
    </tr>
    <tr>
        <td> Silk identity resolution </td>
        <td style="text-align:right;">15:27:52</td>
    </tr>
    <tr>
        <td> Find sameAs URI sets </td>
        <td style="text-align:right;">1:44:24 </td>
    </tr>
    <tr>
        <td> URIs rewriting </td>
        <td style="text-align:right;">1:45:29</td>
    </tr>
    <tr>
        <td> Overall execution  </td>
        <td style="text-align:right;">29:09:03</td>
    </tr>
</table>






<br/><br/> <hr/>

<p id="fn1">
    [1] According to <a href="https://huanliu.wordpress.com/2010/06/14/amazons-physical-hardware-and-ec2-compute-unit/">this article</a>,
    c1.medium instances with 5 ECU are observed to run as 2 of 4 cores of an <a href="http://ark.intel.com/products/33080/Intel-Xeon-Processor-E5410-%2812M-Cache-2_33-GHz-1333-MHz-FSB%29">Intel E5410 processor</a> (4 cores, 2.33 GhZ).
</p>


<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-36110461-1']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>
</body>

</html>