Index: src/com/ontologycentral/ldspider/http/LookupThread.java
===================================================================
--- src/com/ontologycentral/ldspider/http/LookupThread.java	(revision 285)
+++ src/com/ontologycentral/ldspider/http/LookupThread.java	(working copy)
@@ -55,7 +55,7 @@
 	}
 	
 	public void run() {
-		_log.info("starting thread ...");
+		_log.fine("starting thread ...");
 		
 		int i = 0;
 
@@ -145,13 +145,13 @@
 								status = 0;
 							}
 						} else {
-							_log.info("HttpEntity for " + lu + " is null");
+							_log.fine("HttpEntity for " + lu + " is null");
 						}
 					} else if (status == HttpStatus.SC_MOVED_PERMANENTLY || status == HttpStatus.SC_MOVED_TEMPORARILY || status == HttpStatus.SC_SEE_OTHER) { 
 						// treating all redirects the same but shouldn't: 301 -> rename context URI, 302 -> keep original context URI, 303 -> spec inconclusive
 						Header[] loc = hres.getHeaders("location");
 						String path = loc[0].getValue();
-						_log.info("redirecting (" + status + ") to " + path);
+						_log.fine("redirecting (" + status + ") to " + path);
 						URI to = new URI(path);
 						
 						// handle local redirects
@@ -188,6 +188,6 @@
 			lu = _q.poll();
 		}
 		
-		_log.info("finished thread after fetching " + i + " uris");
+		_log.fine("finished thread after fetching " + i + " uris");
 	}
 }
\ No newline at end of file
Index: src/com/ontologycentral/ldspider/http/internal/ResponseGzipUncompress.java
===================================================================
--- src/com/ontologycentral/ldspider/http/internal/ResponseGzipUncompress.java	(revision 285)
+++ src/com/ontologycentral/ldspider/http/internal/ResponseGzipUncompress.java	(working copy)
@@ -70,7 +70,7 @@
             HeaderElement[] codecs = ceheader.getElements();
             for (int i = 0; i < codecs.length; i++) {
                 if (codecs[i].getName().equalsIgnoreCase(GZIP_CODEC)) {
-                	_log.info("gzip compression");
+                	_log.fine("gzip compression");
                     response.setEntity(new GzipDecompressingEntity(response.getEntity())); 
                     return;
                 }
Index: src/com/ontologycentral/ldspider/http/internal/CloseIdleConnectionThread.java
===================================================================
--- src/com/ontologycentral/ldspider/http/internal/CloseIdleConnectionThread.java	(revision 285)
+++ src/com/ontologycentral/ldspider/http/internal/CloseIdleConnectionThread.java	(working copy)
@@ -17,15 +17,15 @@
 		_cm = cm; 
 		_st = sleepTime;
 		
-		log.info("Initialised "+CloseIdleConnectionThread.class.getSimpleName()+" with sleepTime "+_st+" ms");
+		log.fine("Initialised "+CloseIdleConnectionThread.class.getSimpleName()+" with sleepTime "+_st+" ms");
 	}
 
 	public void run() {
-		log.info("Starting "+CloseIdleConnectionThread.class.getSimpleName());
+		log.fine("Starting "+CloseIdleConnectionThread.class.getSimpleName());
 		_run = true;
 		
 		while(_run) {
-			log.info("Closing expired and idle connections");
+			log.fine("Closing expired and idle connections");
 			_cm.closeExpiredConnections();
 			_cm.closeIdleConnections(0L, TimeUnit.SECONDS);
 
@@ -38,12 +38,12 @@
 			}
 		}
 		
-		log.info("Stopped "+CloseIdleConnectionThread.class.getSimpleName());
+		log.fine("Stopped "+CloseIdleConnectionThread.class.getSimpleName());
 	}
 	
 	public void shutdown() {
 		_run = false;
-		log.info("Stopping "+CloseIdleConnectionThread.class.getSimpleName());
+		log.fine("Stopping "+CloseIdleConnectionThread.class.getSimpleName());
 		interrupt();
 	}
 }
Index: src/com/ontologycentral/ldspider/Crawler.java
===================================================================
--- src/com/ontologycentral/ldspider/Crawler.java	(revision 285)
+++ src/com/ontologycentral/ldspider/Crawler.java	(working copy)
@@ -270,7 +270,7 @@
 			
 			m.shutdown();
 			
-			_log.info("ROUND " + curRound + " DONE with " + _queue.size() + " uris remaining in queue");
+			_log.info("round " + curRound + " done - " + _queue.size() + " uris remaining in queue");
 			_log.fine("old queue: \n" + _queue.toString());
 
 			_queue.schedule(frontier);
Index: src/com/ontologycentral/ldspider/hooks/links/LinkFilterDefault.java
===================================================================
--- src/com/ontologycentral/ldspider/hooks/links/LinkFilterDefault.java	(revision 285)
+++ src/com/ontologycentral/ldspider/hooks/links/LinkFilterDefault.java	(working copy)
@@ -53,7 +53,7 @@
 	}
 	
 	public void endDocument() {
-		_log.info("document done");
+		_log.fine("document done");
 	}
 
 	public synchronized void processStatement(Node[] nx) {
@@ -118,7 +118,8 @@
 				if (add) {
 					_f.add(u);
 					_log.fine("adding " + nx[i].toString() + " to frontier");
-					_eh.handleLink(nx[nx.length-1], nx[i]);
+					// Error handler is not defined
+					//_eh.handleLink(nx[nx.length-1], nx[i]);
 				}
 			} catch (URISyntaxException e) {
 				try {
Index: src/com/ontologycentral/ldspider/queue/LoadBalancingQueue.java
===================================================================
--- src/com/ontologycentral/ldspider/queue/LoadBalancingQueue.java	(revision 285)
+++ src/com/ontologycentral/ldspider/queue/LoadBalancingQueue.java	(working copy)
@@ -56,7 +56,7 @@
 	 * @param maxuris - cut off number of uris per pld
 	 */
 	public synchronized void schedule(Frontier f) {	
-		_log.info("start scheduling depth " + _depth++ + "...");
+		_log.fine("start scheduling depth " + _depth++ + "...");
 
 		long time = System.currentTimeMillis();
 
@@ -79,7 +79,7 @@
 		
 		_mintime = _maxtime = System.currentTimeMillis();
 		
-		_log.info("scheduling depth " + _depth + " with " + size() + " uris and " + getSortedQueuePlds().size() + " plds done in " + (_mintime - time) + " ms");
+		_log.fine("scheduling depth " + _depth + " with " + size() + " uris and " + getSortedQueuePlds().size() + " plds done in " + (_mintime - time) + " ms");
 	}
 	
 //	/**
@@ -145,7 +145,7 @@
 			if (_current.isEmpty()) {
 				// queue is empty, done for this round
 				if (size() == 0) {
-					_log.info("queue size is 0: " + toString());
+					_log.fine("queue size is 0: " + toString());
 					return null;
 				}
 				if (_current == POISON) {
@@ -153,19 +153,19 @@
 				}
 		
 				if ((time1 - _mintime) < _mindelay) {
-					_log.info("fetching plds too fast, rescheduling, remaining uris in queue " + size());
-					_log.info(toString());
+					_log.fine("fetching plds too fast, rescheduling, remaining uris in queue " + size());
+					_log.fine(toString());
 					_current = POISON;
 					return null;
 				}
 				
-				_log.info("queue turnaround in " + (time1-_mintime) + " ms");
+				_log.fine("queue turnaround in " + (time1-_mintime) + " ms");
 
 				_mintime = _maxtime = System.currentTimeMillis();
 				
 				_current.addAll(getSortedQueuePlds());
 			} else if ((time1 - _maxtime) > _maxdelay) {
-				_log.info("skipped to start of queue in " + (time1-_maxtime) + " ms, queue size " + size());
+				_log.fine("skipped to start of queue in " + (time1-_maxtime) + " ms, queue size " + size());
 
 				_maxtime = System.currentTimeMillis();
 				
@@ -187,7 +187,7 @@
 		
 		long time1 = System.currentTimeMillis();
 		
-		_log.info("poll for " + next + " done in " + (time1 - time) + " ms");
+		_log.fine("poll for " + next + " done in " + (time1 - time) + " ms");
 
 		return next;
 	}
Index: src/com/ontologycentral/ldspider/queue/BreadthFirstQueue.java
===================================================================
--- src/com/ontologycentral/ldspider/queue/BreadthFirstQueue.java	(revision 285)
+++ src/com/ontologycentral/ldspider/queue/BreadthFirstQueue.java	(working copy)
@@ -60,7 +60,7 @@
 	 * @param maxuris - cut off number of uris per pld
 	 */
 	public synchronized void schedule(Frontier f) {	
-		_log.info("start scheduling...");
+		_log.fine("start scheduling...");
 
 		long time = System.currentTimeMillis();
 		
@@ -107,7 +107,7 @@
 
 		// maxplds means keep only the max number of plds with the largest amount of uris
 		List<String> lipld = getSortedQueuePlds();
-		_log.debug("sorted pld list " + lipld.toString());
+		_log.fine("sorted pld list " + lipld.toString());
 		
 		if (_maxplds != -1) {
 			for (int i = _maxplds; i < lipld.size(); i++) {
@@ -129,8 +129,8 @@
 		}
 		_time = System.currentTimeMillis();
 		
-		_log.debug("scheduling " + _current.size() + " plds done in " + (_time - time) + " ms");
-		_log.debug(toString());
+		_log.fine("scheduling " + _current.size() + " plds done in " + (_time - time) + " ms");
+		_log.fine(toString());
 	}
 		
 	/**
@@ -165,14 +165,14 @@
 							
 				if ((time1 - _time) < CrawlerConstants.MIN_DELAY) {
 					try {
-						_log.info("delaying queue " + CrawlerConstants.MIN_DELAY + " ms ...");
+						_log.fine("delaying queue " + CrawlerConstants.MIN_DELAY + " ms ...");
 						Thread.sleep(CrawlerConstants.MIN_DELAY);
 					} catch (InterruptedException e) {
 						e.printStackTrace();
 					}
 				}
 				
-				_log.info("queue turnaround in " + (time1-_time) + " ms");
+				_log.fine("queue turnaround in " + (time1-_time) + " ms");
 
 				_time = System.currentTimeMillis();
 
Index: src/com/ontologycentral/ldspider/queue/SpiderQueue.java
===================================================================
--- src/com/ontologycentral/ldspider/queue/SpiderQueue.java	(revision 285)
+++ src/com/ontologycentral/ldspider/queue/SpiderQueue.java	(working copy)
@@ -73,7 +73,7 @@
 		}
 		
 		if (from.equals(to)) {
-			_log.info("redirected to same uri " + from);
+			_log.fine("redirected to same uri " + from);
 			return;
 		}
 		
@@ -81,7 +81,7 @@
 //		_redirsRound.add(to);
 		
 		if (checkSeen(to) == false) {
-			_log.info("adding " + to + " directly to queue");
+			_log.fine("adding " + to + " directly to queue");
 			addDirectly(to);
 		}
 	}
@@ -98,7 +98,7 @@
 	URI obtainRedirect(URI from) {
 		URI to = _redirs.getRedirect(from);
 		if (from != to) {
-			_log.info("redir from " + from + " to " + to);
+			_log.fine("redir from " + from + " to " + to);
 			return to;
 		}
 		
Index: src/com/ontologycentral/ldspider/Monitor.java
===================================================================
--- src/com/ontologycentral/ldspider/Monitor.java	(revision 285)
+++ src/com/ontologycentral/ldspider/Monitor.java	(working copy)
@@ -25,7 +25,7 @@
 	public void run() {
 		while (_stop != true) {
 			for (Thread t : _threads) {
-				_pw.println(t.getName());
+				//_pw.println(t.getName());
 			}
 
 			try {
Index: src/com/ontologycentral/ldspider/CrawlerConstants.java
===================================================================
--- src/com/ontologycentral/ldspider/CrawlerConstants.java	(revision 285)
+++ src/com/ontologycentral/ldspider/CrawlerConstants.java	(working copy)
@@ -31,7 +31,7 @@
 
 	public static final int MAX_CONNECTIONS_PER_THREAD = 32;
 	
-	public static final int RETRIES = 0;
+	public static final int RETRIES = 3;
 
 	public static final int MAX_REDIRECTS = 1;
 
@@ -41,7 +41,7 @@
 	public static final int DEFAULT_NB_URIS = Integer.MAX_VALUE;
 	
 	// avoid hammering plds
-	public static final long MIN_DELAY = 2*1000;
+	public static final long MIN_DELAY = 2*100;
 	// for bfs queue: max time after plds get re-visited
 	public static final long MAX_DELAY = 2*MIN_DELAY;
 	
