Index: src/com/ontologycentral/ldspider/http/LookupThread.java
===================================================================
--- src/com/ontologycentral/ldspider/http/LookupThread.java	(revision 281)
+++ src/com/ontologycentral/ldspider/http/LookupThread.java	(working copy)
@@ -172,7 +172,7 @@
 					hget.abort();
 				} catch (Throwable e) {
 					hget.abort();
-					_log.warning("Exception " + e.getClass().getName());
+					_log.warning("Exception " + e.getClass().getName() +"\n"+ e.getMessage());
 					_eh.handleError(lu, e);
 				}
 				
Index: src/com/ontologycentral/ldspider/hooks/links/LinkFilterDefault.java
===================================================================
--- src/com/ontologycentral/ldspider/hooks/links/LinkFilterDefault.java	(revision 281)
+++ src/com/ontologycentral/ldspider/hooks/links/LinkFilterDefault.java	(working copy)
@@ -118,7 +118,8 @@
 				if (add) {
 					_f.add(u);
 					_log.fine("adding " + nx[i].toString() + " to frontier");
-					_eh.handleLink(nx[nx.length-1], nx[i]);
+					// Error handler is not defined
+					//_eh.handleLink(nx[nx.length-1], nx[i]);
 				}
 			} catch (URISyntaxException e) {
 				try {
